geopy/geocoders/navidata.py

```diff
@@ -0,0 +1,195 @@
+    """
+    Geocoder using the NaviData  API.
+    See http://www.navidata.pl
+    """
+
+    def __init__(
+            self,
+            api_key=None,
+            domain='api.navidata.pl',
+            timeout=DEFAULT_TIMEOUT,
+            proxies=None,
+    ):
+        """
+        Initialize NaviData geocoder. Please note that 'scheme' parameter is not supported - at present state, all
+        NaviData traffic use plain http.
+
+        :param string api_key: The commercial API key for service. None required if you
+            use this API for non-commercial puproses
+
+        :param string domain: Currently it is 'api.navidata.pl', can
+            be changed for testing purposes.
+
+        :param dict proxies: If specified, routes this geocoder's requests
+            through the specified proxy. E.g., {"https": "192.0.2.0"}. For
+            more information, see documentation on
+            :class:`urllib2.ProxyHandler`.
+
+        """
+        super(NaviData, self).__init__(
+            scheme="http", timeout=timeout, proxies=proxies
+        )
+
+        self.api_key = api_key
+        self.domain = domain.strip('/')
+        #self.scheme = scheme
+        self.geocode_api = 'http://%s/geocode' % (self.domain)
+        self.reverse_geocode_api = 'http://%s/revGeo' % (self.domain)
+
+    def geocode(
+            self,
+            query,
+            exactly_one=True,
+            timeout=None,
+    ):
+        """
+        Geocode a location query.
+
+        :param string query: The query string to be geocoded; this must
+            be URL encoded.
+
+        :param bool exactly_one: Return one result or a list of results, if
+            available.
+
+        :param int timeout: Time, in seconds, to wait for the geocoding service
+            to respond before raising a :class:`geopy.exc.GeocoderTimedOut`
+            exception. Set this only if you wish to override, on this call
+            only, the value set during the geocoder's initialization.
+
+        """
+        params = {
+            'q': self.format_string % query,
+        }
+
+
+        if self.api_key is not None:
+            params["api_key"] = self.api_key
+
+        url = "?".join((self.geocode_api, urlencode(params)))
+
+        logger.debug("%s.geocode: %s", self.__class__.__name__, url)
+        return self._parse_json_geocode(
+            self._call_geocoder(url, timeout=timeout), exactly_one
+        )
+
+    def reverse(
+            self,
+            query,
+            exactly_one=False,
+            timeout=None,
+    ):
+
+        """
+        Given a point, find an address.
+
+        :param query: The coordinates for which you wish to obtain the
+            closest human-readable addresses.
+        :type query: :class:`geopy.point.Point`, list or tuple of (latitude,
+            longitude), or string as "%(latitude)s, %(longitude)s"
+
+        :param boolean exactly_one: Return one result or a list of results, if
```

Comment:
Is `exactly_one` only invalid for `reverse`? In any case, I think the default value should be `True` then.